### DISCUSSION TITLE
Collection Content create fails in KnowledgeMW

### DISCUSSION THREAD

**Discussion by tsprasath** [Sunbird-Knowlg/Community#130] [General Discussions]
Hi Team,

**steps:**
After portal login->navigate to workspace-->select collection --> and select **Content Playlist** after clicking submit
content create fails

Similar way for **Digital Textbook** and **question paper**
`{"id":"083572a6-dcbc-eb84-a7b5-a5c2f28a93da","msg":"Getting error f
rom content provider while creating content","err":{"err":null,"errCode":"404","errMsg":"Entry is not found
 in cassandra for content with identifier: obj-cat:content-playlist_collection_all {}","responseCode":"RESO
URCE_NOT_FOUND"},"additionalInfo":{"ekStepReqData":{"request":{"content":{"primaryCategory":"Content Playli
st","name":"V12","description":"Enter description for Collection","createdBy":"4f2bcb07-6d8e-4022-9866-6e54
f1222514","organisation":["Sunbird Org"],"createdFor":["0137972723324682240"],"framework":"sunbird_k-12","m
imeType":"application/vnd.ekstep.content-collection","resourceType":"Collection","creator":"SunbirdOrgAdmin
4","contentType":"Collection","code":"org.sunbird.HQDSLG"}}}}}`
``
[@maheshkumargangula](https://github.com/maheshkumargangula), [@vrayulu](https://github.com/vrayulu), @Krishnaj20,[@pallakartheekreddy](https://github.com/pallakartheekreddy), [@gandham-santhosh](https://github.com/gandham-santhosh), [@rajeevsathish](https://github.com/rajeevsathish)

**Comment by pallakartheekreddy** [comment]
@tsprasath From above error it is clearly stating, you are trying to create **Content Playlist** from UI without creating Content Playlist primaryCategory. Please first create Content Playlist primaryCategory object category definition APIs.
Ref URL: https://knowlg.sunbird.org/learn/product-and-developer-guide/taxonomy-and-tagging/object-category-service/apis

**Reply by tsprasath** [reply]
After executing the below curl issue got sorted.

Below command will create object category for Digital TextBook

```
curl --location -g --request POST '{{host}}/api/object/category/v1/create' \
--header 'Content-Type: application/json' \
--header 'Authorization: {{kong_api_key}}' \
--data-raw '{
    "request": {
        "objectCategory": {
            "name": "Digital Textbook",
            "description": "Digital Textbook"
        }
    }
}'
```

For verification read the category definition 

```
curl --location -g --request POST '{{host}}/api/object/category/definition/v1/read?fields=objectMetadata,forms,name' \
--header 'Authorization: {{kong_api_key}}' \
--header 'Content-Type: application/json' \
--data-raw '{
    "request": {
        "objectCategoryDefinition": {
            "objectType": "Collection",
            "name": "Digital Textbook"
        }
    }
}'
```
